R/test-that.R

```diff
@@ -13,6 +13,46 @@
 #' changes. Avoid changing global state, when possible, and reverse any changes
 #' that you do make.
 #'
+#' # Detecting changes to global state
+#'
+#' One of the most pernicious challenges to debug is when a test runs fine
+#' in your test suite, but fails when you run it interactively (or similarly,
+#' it fails randomly when running your tests in parallel). One of the most
+#' common causes of this problem is accidentally changing global state in a
+#' previous test (e.g. changing an option, an environment variable, or the
+#' working directory). This is hard to debug, because it's very hard to figure
+#' out which test made the change.
+#'
+#' Luckily testthat provides a tool to figure out if tests are changing global
+#' state. You can write a function called `testthat_state` and
```

Comment:
```suggestion
#' state. You can write a function called `testthat_state()` and
```

The phrasing ("called `testthat_state`") makes it hard to know whether the parens are appropriate, but I still think so. Or re-phrase.